src/inmanta/module.py

```diff
@@ -301,7 +301,7 @@ def make_repo(path: str, root: Optional[str] = None) -> Union[LocalFileRepo, Rem
 def merge_specs(mainspec: "Dict[str, List[Requirement]]", new: "List[Requirement]") -> None:
     """Merge two maps str->[T] by concatting their lists."""
     for req in new:
-        key = req.project_name
+        key = req.project_name.replace("-", "_")
```

Comment:
```suggestion
        key = req.key
```
I think this may be what you're looking for.

Reply:
Do you mean `req.key.replace("-", "_")`? Because `req.key` constained the name of the requirement in standard notation, but the standard notation consists of `-` characters and not `_` characters.

Reply:
Right, and we use this dict to do module lookup, you're right.